adapters/core2p2p/block.go

```diff
@@ -0,0 +1,40 @@
+package core2p2p
+
+import (
+	"time"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/p2p/starknet/spec"
+	"github.com/NethermindEth/juno/utils"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func AdaptBlockID(header *core.Header) *spec.BlockID {
+	return &spec.BlockID{
+		Number: header.Number,
+		Header: AdaptHash(header.Hash),
+	}
+}
+
+func AdaptHeader(header *core.Header) *spec.BlockHeader {
+	return &spec.BlockHeader{
+		ParentHeader:     AdaptHash(header.ParentHash),
+		Number:           header.Number,
+		Time:             timestamppb.New(time.Unix(int64(header.Timestamp), 0)),
+		SequencerAddress: AdaptAddress(header.SequencerAddress),
+		StateDiffs:       nil,
+		State:            nil,
+		ProofFact:        nil, // ???
+		Transactions:     nil,
+		Events:           nil,
+		Receipts:         nil,
```

Comment:
Where should I get those ? 😭

Reply:
Does it work if we get those values from the db and pass them into the adapter?

Reply:
Sure, I just need to know which values to use for field that are nil